lib/march_hare/session.rb

```diff
@@ -180,6 +180,7 @@ def initialize(connection_factory, opts = {})
       @shutdown_hooks            = Array.new
       @blocked_connection_hooks  = Array.new
       @connection_recovery_hooks = Array.new
+      @session_closed            = false
```

Comment:
I think `@was_explicitly_closed` would be more to the point.

Reply:
👍 will change the variable name